### ISSUE TITLE
`expire_after` with value -1 in `session.request` doesn't override `CachedSession.expire_after`

### ISSUE THREAD

**Issue by johnraz** [reclosedev/requests-cache#526]
### The problem
When specifying `expire_after=-1` to the `session.request()` method the response is not cached forever but it falls back to the value set on the `CachedSession` instead.

### Expected behavior
When specifying `expire_after=-1` to the `session.request()` method I would expect the response to be cached forever, overriding anything that was set on the `CachedSession.expire_after`.

### Steps to reproduce the behavior
```
>>> from requests_cache import CachedSession
>>> session = CachedSession(in_memory=True, expire_after=10)

>>> response = session.get('https://httpbin.org/get', expire_after=-1)
>>> print(response)
<Response [200]>

>>> response = session.get('https://httpbin.org/get', expire_after=-1)

>>> print(response)
request: GET https://httpbin.org/get, response: 200 (350 bytes), created: 2022-02-15 18:23:32 CET, expires: 2022-02-15 18:23:42 CET (fresh)

# A few moment later...
>>> print(response)
request: GET https://httpbin.org/get, response: 200 (350 bytes), created: 2022-02-15 18:23:32 CET, expires: 2022-02-15 18:23:42 CET (stale)

# A few moment later...
>>> response = session.get('https://httpbin.org/get', expire_after=-1)

>>> print(response)
<Response [200]>
```

### Workarounds
I couldn't find any but I believe the bug is caused by this code here:
```
def get_expiration_datetime(expire_after: ExpirationTime) -> Optional[datetime]:
    """Convert an expiration value in any supported format to an absolute datetime"""
    # Never expire
    if expire_after is None or expire_after == -1:
        return None
   ...
```

Once `get_expiration_datetime` returns None the code will continue and assume that no `expire_after` was set and default on the one available on the `CachedSession`

### Environment
- requests-cache version: `0.9.1`
- Python version: `3.8.12`
- Platform: MacOS

**Comment by JWCook** [comment]
Thanks for the bug report! I'll get that fixed soon.

**Comment by johnraz** [comment]
@JWCook  thanks so much for replying so quickly and btw very neat library, I really like it so far 🙂

**Comment by JWCook** [comment]
Fixed in `master`. I'll include this in a patch release either today or later this week.

**Comment by JWCook** [comment]
This is now fixed in v0.9.2.

**Comment by johnraz** [comment]
@JWCook  this is awesome!! thanks so much again!
